lib/libc/common/source/stdio/getc_unlocked.c

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2024 Meta Platforms
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <stdio.h>
+
+#include <zephyr/sys/util.h>
+
+int getc_unlocked(FILE *stream)
+{
+	ARG_UNUSED(stream);
+
+	errno = ENOSYS;
+
+	return EOF;
+}
```

Comment:
For this one, I would add a `zvfs_getc_unlocked()` and just call that.

`zvfs_getc_unlocked()` should do some basic error checking but should not incur lock overhead.

E.g. check to see if `stream` is a valid pointer with `IS_ARRAY_ELEMENT()`, check the `read()` vtable method is non-`NULL`, etc.